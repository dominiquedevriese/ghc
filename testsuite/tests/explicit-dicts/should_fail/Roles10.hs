{-# LANGUAGE RoleAnnotations #-}
module Roles10 where

data Set a
type role Set nominal

class SetInsertable a where
  insert :: a -> Set a -> Set a

foo :: SetInsertable a => a -> a
foo x = x


reverseSetInsertable :: SetInsertable.Dict Int
reverseSetInsertable = undefined

f10 = foo ((reverseSetInsertable))
