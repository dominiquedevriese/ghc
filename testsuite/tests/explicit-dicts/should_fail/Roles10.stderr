
Roles10.hs:17:7: error:
    • Explicit dictionary application to: ‘SetInsertable a’
      is not allowed because of instance incoherence:
      In the dictionary type ‘SetInsertable.Dict a’
        Type variable ‘a’ has role Nominal
    • In the expression: foo ((reverseSetInsertable))
      In an equation for ‘f10’: f10 = foo ((reverseSetInsertable))
